src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -225,4 +229,84 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
+        if (isExistingPaymentWorksVendor(kfsDocumentNumber) && isMaintenanceActionNewOrNewWithExisting(maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: isExistingPaymentWorksVendor");
+            this.saveBusinessObject();
+            processExistingPaymentWorksVendorApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (isMaintenanceActionNewOrNewWithExisting(maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: new action resulting from AVf or hand keying");
+            this.saveBusinessObject();
+            processKfsVendorNewActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (StringUtils.equals(KRADConstants.MAINTENANCE_EDIT_ACTION, maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: edit action");
+            processKfsVendorEditActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+    }
+    
+    private boolean isMaintenanceActionNewOrNewWithExisting(String maintenanceAction) {
+        return (StringUtils.equals(KRADConstants.MAINTENANCE_NEW_ACTION, maintenanceAction)
+                || StringUtils.equals(KRADConstants.MAINTENANCE_NEWWITHEXISTING_ACTION, maintenanceAction));
+    }
+    
+    private boolean isExistingPaymentWorksVendor(String kfsDocumentNumber) {
+        return getPaymentWorksBatchUtilityService().foundExistingPaymentWorksVendorByKfsDocumentNumber(kfsDocumentNumber);
+    }
+    
+    private boolean recordPvenRouteStatusChangeForPaymentWorksIsEnabled() {
+        return (getPaymentWorksBatchUtilityService().retrievePaymentWorksParameterValue(PaymentWorksParameterConstants.PMW_INTEGRATION_IS_ACTIVE_IND).equalsIgnoreCase(KFSConstants.ParameterValues.YES));
+    }
+    
+    private void processExistingPaymentWorksVendorApproval(String kfsDocumentNumber, VendorDetail vendorDetail) {
+        getPaymentWorksBatchUtilityService().registerKfsPvenApprovalForExistingPaymentWorksVendor(kfsDocumentNumber, vendorDetail);
+    }
+    
+    private void processKfsVendorNewActionApproval(String kfsDocumentNumber, VendorDetail vendorDetail) {
+        getPaymentWorksBatchUtilityService().registerKfsPvenApprovalForKfsEnteredVendor(kfsDocumentNumber, vendorDetail);
+    }
+    
+    private void processKfsVendorEditActionApproval(String kfsDocumentNumber, VendorDetail vendorDetail) {
+        getPaymentWorksBatchUtilityService().registerKfsPvenApprovalForKfsEditedVendor(kfsDocumentNumber, vendorDetail);
+    }
+    
+    private void performPaymentWorksDisapprovalCancelProcessingForVendor(VendorDetail vendorDetail, String kfsDocumentNumber, String maintenanceAction) {
+        if (isExistingPaymentWorksVendor(kfsDocumentNumber) && (isMaintenanceActionNewOrNewWithExisting(maintenanceAction))) {
+            getPaymentWorksBatchUtilityService().registerKfsPvenDisapprovalForExistingPaymentWorksVendor(kfsDocumentNumber, vendorDetail);
+        }
```

Comment:
Is a plain "else" block also needed here to at least log the cases where no PaymentWorks processing is needed?

Reply:
Same message as above.